stack.go

```diff
@@ -6,47 +6,80 @@ import (
 	"strings"
 )
 
+// Stack is an array of stack frames stored in a human readable format.
+type Stack []StackFrame
+
+// contains checks if a stack trace contains a specific frame.
+func (s *Stack) contains(frame StackFrame) bool {
+	for _, f := range *s {
```

Comment:
Make sure this can't be a nil ptr

Reply:
I verified if nil is passed then segfault is guaranteed

Reply:
cool thanks for testing that, i'll fix and try to add a test case for it